runtime/queries/column_timeseries.go

```diff
@@ -0,0 +1,537 @@
+	r, err := json.Marshal(q)
+	if err != nil {
+		panic(fmt.Errorf("ColumnTimeseries: failed to marshal: %w", err))
+	}
+	return fmt.Sprintf("ColumnTimeseries:%s", string(r))
+}
+
+func (q *ColumnTimeseries) Deps() []string {
+	return []string{q.TableName}
+}
+
+func (q *ColumnTimeseries) MarshalResult() any {
+	return q.Result
+}
+
+func (q *ColumnTimeseries) UnmarshalResult(v any) error {
+	res, ok := v.(*runtimev1.TimeSeriesResponse)
+	if !ok {
+		return fmt.Errorf("ColumnTimeseries: mismatched unmarshal input")
+	}
+	q.Result = res
+	return nil
+}
+
+func (q *ColumnTimeseries) Resolve(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int) error {
+	olap, err := rt.OLAP(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+
+	if olap.Dialect() != drivers.DialectDuckDB {
+		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
+	}
+
+	timeRange, err := q.normaliseTimeRange(ctx, rt, instanceID, priority)
+	if err != nil {
+		return err
+	}
+	if timeRange.Interval == runtimev1.TimeGrain_TIME_GRAIN_UNSPECIFIED {
+		q.Result = &runtimev1.TimeSeriesResponse{}
+		return nil
+	}
+	var measures = normaliseMeasures(q.Measures)
+	var timestampColumn = q.TimestampColumnName
+	var tableName = q.TableName
+	var filter string
+	if q.Filters != nil {
+		filter = getFilterFromMetricsViewFilters(q.Filters)
+	}
+	var timeGranularity = convertToDateTruncSpecifier(timeRange.Interval)
+	var tsAlias string
+	if timestampColumn == "ts" {
+		tsAlias = "_ts"
+	} else {
+		tsAlias = "ts"
+	}
+	if filter != "" {
+		filter = "WHERE " + filter
+	}
+	temporaryTableName := "_ts_" + uuid.New().String()
+	sql := `CREATE TEMPORARY TABLE "` + temporaryTableName + `" AS (
+        -- generate a time series column that has the intended range
+        WITH template as (
+          SELECT 
+            generate_series as ` + tsAlias + `
+          FROM 
+            generate_series(
+              date_trunc('` + timeGranularity + `', TIMESTAMP '` + timeRange.Start.AsTime().Format(IsoFormat) + `'),
+              date_trunc('` + timeGranularity + `', TIMESTAMP '` + timeRange.End.AsTime().Format(IsoFormat) + `'),
+              interval '1 ` + timeGranularity + `')
+        ),
+        -- transform the original data, and optionally sample it.
+        series AS (
+          SELECT 
+            date_trunc('` + timeGranularity + `', "` + EscapeDoubleQuotes(timestampColumn) + `") as ` + tsAlias + `,` + getExpressionColumnsFromMeasures(measures) + `
+          FROM "` + EscapeDoubleQuotes(tableName) + `" ` + filter + `
+          GROUP BY ` + tsAlias + ` ORDER BY ` + tsAlias + `
+        )
+        -- join the transformed data with the generated time series column,
+        -- coalescing the first value to get the 0-default when the rolled up data
+        -- does not have that value.
+        SELECT 
+          ` + getCoalesceStatementsMeasures(measures) + `,
+          template.` + tsAlias + ` as ts from template
```

Comment:
Is it safe to select this using `as ts` given that a measure might be named `ts`? Shouldn't it also use `tsAlias`?

Reply:
Yes, it will be safer.